cloudinit/sources/DataSourceAzure.py

```diff
@@ -554,6 +554,38 @@ def _is_ephemeral_networking_up(self) -> bool:
             or self._ephemeral_dhcp_ctx.lease is None
         )
 
+    def _check_azure_proxy_agent_status(self) -> None:
+        try:
+            cmd = [
+                "azure-proxy-agent",
+                "--status",
+                "--wait",
+                "120",
+            ]
+            out, err = subp.subp(cmd)
+            report_diagnostic_event(
+                "Running azure-proxy-agent",
+                logger_func=LOG.info,
+            )
+            if err:
+                report_diagnostic_event(
```

Comment:
consolidate this with the above diagnostic

Reply:
done